src/components/ComboBox.jsx

```diff
@@ -592,13 +589,13 @@ export default class ComboBox extends Component {
    */
   static filterOptions(inputValue, properties) { /*eslint react/sort-comp:0*/
     let filteredOptions = [];
-    if (properties.children.length > 0) {
+    if (!isEmpty(properties.children)) {
       if (inputValue) {
-        filteredOptions = filter(properties.children, (entry) => {
+        filteredOptions = filterReactChildren(properties.children, (entry) => {
           return properties.filterFunc(inputValue, entry.props.value);
         });
       } else {
-        filteredOptions = map(properties.children, (entry) => { return entry; });
+        filteredOptions = React.children.map(properties.children, (entry) => { return entry; });
```

Comment:
props.children can be anything array or object, filteredOptions I have always made as array.
I could have created even that as props.children and used React.children utility methods for it, but across comboBox code there is a lot of position access in filteredOptions using index which will be most optimum with Array.